src/hooks/useDarkMode.ts

```diff
@@ -3,18 +3,28 @@ import { selectSettings } from '@/store/settingsSlice'
 import { useEffect, useMemo, useState } from 'react'
 import initTheme from '@/styles/theme'
 
-const mediaQuery = '(prefers-color-scheme: dark)'
+const isSystemDarkMode = (): boolean => {
+  const mediaQuery = '(prefers-color-scheme: dark)'
+  return typeof window !== 'undefined' && window.matchMedia(mediaQuery).matches
+}
 
-export const useDarkMode = () => {
+export const useDarkMode = (): boolean => {
   const settings = useAppSelector(selectSettings)
   const [isDarkMode, setIsDarkMode] = useState<boolean>(false)
 
   useEffect(() => {
-    const systemPreference = window.matchMedia(mediaQuery)
-    setIsDarkMode(settings.theme.darkMode ?? systemPreference.matches)
+    setIsDarkMode(settings.theme.darkMode ?? isSystemDarkMode())
+  }, [settings.theme.darkMode])
+
+  return isDarkMode
+}
 
+export const useDarkModeTheme = () => {
```

Comment:
Nit: I think `useTheme` or something similar would be a bit better, although MUI has a `useTheme` hook already.

Reply:
`useLightDarkTheme`?

Reply:
Sounds good to me!